aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -194,6 +196,10 @@ module aptos_framework::fungible_asset {
         metadata: Object<Metadata>
     }
 
+    struct WithdrawPermission has copy, drop, store {
+        metadata_address: address,
```

Comment:
I think making it an enum is a great idea. However, I would avoid adding `store_addr` at the moment as this would require coordination with wallet side how the permission should be displayed.